### PR TITLE
Form 8 all questions

### PR DESCRIPTION
**Pull request by webinista**
- Adds remaining questions from Form 8
     - Adds a partial to accommodate questions with multiple radio buttons
     - Adds a `checked` variable to two_radio, yes/no radio button, and multi-radio partials
- Adds a fix for #269, so `legend` elements now wrap.
- Changing some class names to use a `cf-` prefix instead of `usa-cf_app` or being unprefixed. (Helps other devs determine where to start looking for styles.)

### CODE REVIEW

File: app/views/web/questions.html.erb [hunk 1]
```diff
@@ -9,6 +9,130 @@
           <p class="required-msg usa-input-error-message">Required</p>
       </header>
 
+
+     <% if @kase.correspondent.appellant_name %>
```
Comment (paultag) [inline app/views/web/questions.html.erb]: I think we need to always show these, even if the data is `''` -- thoughts @plusjeff ?

File: app/views/web/questions.html.erb [hunk 2]
```diff
@@ -18,33 +142,39 @@
...
                 option2_value: '8B_CERTIFICATION_THAT_VALID_POWER_OF_ATTORNEY_IS_IN_ANOTHER_VA_FILE',
-                required: true,
             } %>
 
         <%= render partial: 'form8_text', locals: {
+                div_class: 'cf-wrap-flex',
                 div_id: '8b-explanation',
-                question: 'Please specify location of the veteran\'s power of attorney',
-                question_num: '8B EXPLANATION',
+                question: '<em>Specify file if applicable</em>',
```
Comment (paultag) [inline app/views/web/questions.html.erb]: @maryannbrody What should we do here?

**Comment by paultag** [comment]
:+1: this is awesome. Killer work, thanks @webinista!

**Comment by plusjeff** [comment]
Taking a peak in IE 11(let me know if someone did this already) and looking at the questions posted above.

File: app/views/web/questions.html.erb (hunk 2 above)
Comment (maryannbrody) [inline app/views/web/questions.html.erb]: @paultag - I think that looks fine!

**Comment by paultag** [comment]
@plusjeff ah, I hadn't yet, was talking about something else; I'll let you do testing and merge! :+1:

File: app/views/web/questions.html.erb (hunk 2 above)
Comment (paultag) [inline app/views/web/questions.html.erb]: Great, thanks!

**Comment by plusjeff** [comment]
Nitpick: Can this get auto-filled with today's date?

image

File: app/views/web/questions.html.erb (hunk 1 above)
Comment (plusjeff) [inline app/views/web/questions.html.erb]: @paultag Agreed

**Comment by plusjeff** [comment]
For the date fields, can we use the `date` input type? (i.e. `<input type="date"...>`)

**Comment by plusjeff** [comment]
Let's not let this field be editable or submitable:

image

**Comment by plusjeff** [comment]
This field should be required:

image

**Comment by plusjeff** [comment]
Can we make these fields non-editable?

image

**Comment by plusjeff** [comment]
9B does not get populated when it has a value

image

**Comment by plusjeff** [comment]
Looks like 10B Yes can't be clicked.

10b

**Comment by plusjeff** [comment]
10C doesn't populate on the PDF when it has a value

image

**Comment by plusjeff** [comment]
Fields 5A, 6A, and 7A appear to not be filling in (tried with `full-form8` veteran)

master branch:
image

Form for this branch:
image

**Comment by plusjeff** [comment]
8C is also not getting auto-filled (also for `full-form8` veteran):

image

**Comment by webinista** [comment]
@plusjeff re: 

> Nitpick: Can this get auto-filled with today's date?

Any preference about whether this is done with JavaScript or Ruby/ERB?

**Comment by plusjeff** [comment]
@webinista RE:  Date

On second thought, I'm not sure this field really has to be editable even if it's visible. It should be the date for when they click submit anyway. :smile:

**Comment by plusjeff** [comment]
^^ In that case, ERB probably works best.

**Comment by webinista** [comment]
@plusjeff 

> For the date fields, can we use the date input type? (i.e. <input type="date"...>)

Yes. Dates need to be formatted in `yyyy-mm-dd` format though. Right now, they're being pulled from the model in `mm/dd/yyyy` format. I can update the value using JavaScript after loading if it's too much work to change the model.

**Comment by plusjeff** [comment]
@webinista I think Rails can do the lifting on that too:

http://edgeapi.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-date_field

**Comment by plusjeff** [comment]
@webinista Aw fooey. Ignore me. IE 11 doesn't support input type date :smile:

**Comment by webinista** [comment]
@plusjeff That can be a low priority addition then. Edge, the latest MS browser supports `type=date`. It'll just behave like a `text` field in IE11. Can you file a bug and add the `enhancement` label?

**Comment by webinista** [comment]
Rounding all of these changes up in one place so I make sure I have them all:

- [ ] _These xB fields don't need to be filled in if the corresponding xA field is blank._ (e.g., if 5A is blank, 5b doesn't need to be filled in.
- [ ] Make sure 8C is autofilled
- [ ] Make 17C required, and auto fill the date.

**Comment by plusjeff** [comment]
Re:

> Can you file a bug and add the enhancement label?

Good idea! Done. #292

**Comment by plusjeff** [comment]
Almost there! :smile: Confirmed the above checklist, but found these (related to the last checkbox above)

- [ ] 9A and 9B don't show up on questions page (see `full-form8` veteran)
- [ ] 10C doesn't make it to the PDF

**Comment by plusjeff** [comment]
I'm testing out a few of the conditional fill-ins too.

**Comment by plusjeff** [comment]
Other conditional fields are good!

**Comment by plusjeff** [comment]
From Slack:

> In `web_controller.rb`, if you replace line 86 with this, the 10C bug should be fixed:
`params.each{|k, v| fields[k] = v }`

**Comment by plusjeff** [comment]
@webinista @paultag Added a couple fixes in. 

@paultag : Can you give it a last look over and test tomorrow?

**Comment by paultag** [comment]
On it

**Comment by paultag** [comment]
Still reviewing; can we land a patch on this like:

```diff
diff --git a/app/views/web/questions.html.erb b/app/views/web/questions.html.erb
index b101b76..22b9e69 100644
--- a/app/views/web/questions.html.erb
+++ b/app/views/web/questions.html.erb
@@ -71,7 +71,7 @@
              question: "Date of notification of action appealed",
              question_num: '5B',
              input_name: '5B_DATE_OF_NOTIFICATION_OF_ACTION_APPEALED',
-             value: @kase.bfdrodec,
+             value: Caseflow.format_issues(@kase, 'service connection') != '' ? @kase.bfdrodec : '',
          } %>
 
     </div>
@@ -90,7 +90,7 @@
              question: "Date of notification of action appealed",
              question_num: '6B',
              input_name: '6B_DATE_OF_NOTIFICATION_OF_ACTION_APPEALED',
-             value: @kase.bfdrodec,
+             value: Caseflow.format_issues(@kase, 'increased rating') != '' ? @kase.bfdrodec : '',
          } %>
     </div>
 
@@ -108,7 +108,7 @@
              question: "Date of notification of action appealed",
              question_num: '7B',
              input_name: '7B_DATE_OF_NOTIFICATION_OF_ACTION_APPEALED',
-             value: @kase.bfdrodec,
+             value: Caseflow.format_issues(@kase, 'other') != '' ? @kase.bfdrodec : '',
          } %>
 
     </div>
```

cc @plusjeff @webinista

**Comment by paultag** [comment]
That way, a user won't have to clear out date fields that don't need to be set -- and if it needs to be filled in later, it's no worse than the current state (until we can more magic it, I guess)

**Comment by paultag** [comment]
Otherwise, LGTM

**Comment by paultag** [comment]
(Kicked back to @webinista for further thought and reactions -- you too, @plusjeff)

**Pull request merged**